workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -810,5 +810,45 @@ for (var j = 0; j < commandDetailsList.Count; j++) {
                 chunkArray.Dispose();
             }
         }
+
+        internal class AcknowledgeAuthorityLossHandler : AbstractAcknowledgeAuthorityLossHandler
+        {
+            public override EntityArchetypeQuery Query => new EntityArchetypeQuery
+            {
+                All = new ComponentType[]
+                {
+                    ComponentType.ReadOnly<AuthorityLossImminent<<#= componentNamespace #>.Component>>(),
+                    ComponentType.ReadOnly<SpatialEntityId>()
+                },
+                Any = Array.Empty<ComponentType>(),
+                None = Array.Empty<ComponentType>()
+            };
+
+            public override void AcknowledgeAuthorityLoss(ComponentGroup group, ComponentSystemBase system,
+                Improbable.Worker.Core.Connection connection)
+            {
+                var authorityLossType = system.GetArchetypeChunkComponentType<AuthorityLossImminent<<#= componentNamespace #>.Component>>();
```

Comment:
Would it help to use `IsEmptyIgnoreFilter` here?

Reply:
We have to AB test to see if that is actually faster than just being returned a 0 length array.

Reply:
Right now I suspect no, although that might change in future.